### PR TITLE
use constants when possible

### PR DESCRIPTION
**Pull request by orklah**
This PR proposes using constants instead of literal string when possible.

This was generated by https://github.com/orklah/psalm-use-constants and then checked by hand. It's possible that some constants could be used where the context does not match but I don't know doctrine's internal well enough to judge every case.

### CODE REVIEW

File: lib/Doctrine/ORM/Cache/QueryCacheEntry.php [hunk 1]
```diff
@@ -58,6 +60,6 @@ public function __construct($result, $time = null)
      */
     public static function __set_state(array $values)
     {
-        return new self($values['result'], $values['time']);
+        return new self($values['result'], $values[Types::TIME_MUTABLE]);
```
Comment (greg0ire) [inline lib/Doctrine/ORM/Cache/QueryCacheEntry.php]: Unrelated I think

File: lib/Doctrine/ORM/Cache/TimestampCacheEntry.php [hunk 2]
```diff
@@ -53,6 +55,6 @@ public function __construct($time = null)
      */
     public static function __set_state(array $values)
     {
-        return new self($values['time']);
+        return new self($values[Types::TIME_MUTABLE]);
```
Comment (greg0ire) [inline lib/Doctrine/ORM/Cache/TimestampCacheEntry.php]: Same

File: lib/Doctrine/ORM/Tools/Console/Command/RunDqlCommand.php [hunk 3]
```diff
@@ -54,7 +55,7 @@ protected function configure()
         $this->setName('orm:run-dql')
              ->setDescription('Executes arbitrary DQL directly from the command line')
              ->addArgument('dql', InputArgument::REQUIRED, 'The DQL to execute.')
-             ->addOption('hydrate', null, InputOption::VALUE_REQUIRED, 'Hydration mode of result set. Should be either: object, array, scalar or single-scalar.', 'object')
+             ->addOption('hydrate', null, InputOption::VALUE_REQUIRED, 'Hydration mode of result set. Should be either: object, array, scalar or single-scalar.', Types::OBJECT)
```
Comment (greg0ire) [inline lib/Doctrine/ORM/Tools/Console/Command/RunDqlCommand.php]: Unrelated: array ,  scalar and single-scalar are not Types constants are they?

File: lib/Doctrine/ORM/Cache/QueryCacheEntry.php (hunk 1 above)
Comment (orklah) [reply lib/Doctrine/ORM/Cache/QueryCacheEntry.php]: reverted :)

**Review by SenseException — DISMISSED** [review]

**Comment by orklah** [comment]
Build failure have to be unrelated but not sure where it came from

**Comment by greg0ire** [comment]
It looks like an issue with the terminal width, which is governed by https://github.com/doctrine/orm/blob/3f3de70c3e7e03e2c2e67fb990eda401b387b5d3/phpunit.xml.dist#L67

My bet is this has to do with the last `symfony/console` version

EDIT: looks like it! https://github.com/symfony/symfony/pull/40348

**Comment by greg0ire** [comment]
Blocked by https://github.com/doctrine/orm/pull/8573

**Comment by greg0ire** [comment]
Ah it needs a merge up

**Comment by greg0ire** [comment]
@orklah there are conflicts, please rebase.

**Comment by orklah** [comment]
Done. Psalm errors seems unrelated

**Comment by greg0ire** [comment]
> Psalm errors seems unrelated

Yep, this is due to my latest work on `doctrine/persistence` I'm afraid. I will look into this, but you can add them to the baseline in the meantime if you want this to move forward. They don't seem to affect 2.8.x, which is kind of weird.

**Comment by orklah** [comment]
Baseline regenerated. Maybe 2.8 is not affected because doctrine/persistence is pinned to an older version?

**Comment by greg0ire** [comment]
It does not seem to be… maybe the strictness is not the same?

**Pull request closed without merging**